src/CSET/operators/imageprocessing.py

```diff
@@ -0,0 +1,278 @@
+       doi: 10.1109/TIP.2003.819861
+
+    Examples
+    --------
+    >>> MSSIM = imageprocessing.structural_similarity_model_comparisons(
+            cubes,sigma=1.5, spatial_plot=False)
+    >>> iplt.plot(MSSIM)
+
+    >>> SSIM = imageprocessing.structural_similarity_model_comparisons(
+            cubes, sigma=1.5, spatial_plot=True)
+    >>> iplt.pcolormesh(SSIM[0,:], cmap=mpl.cm.bwr)
+    >>> plt.gca().coastlines('10m')
+    >>> plt.clim(-1, 1)
+    >>> plt.colorbar()
+    >>> plt.show()
+    """
+    if len(cubes) != 2:
+        raise ValueError("cubes should contain exactly 2 cubes.")
+    base: iris.cube.Cube = cubes.extract_cube(
+        iris.AttributeConstraint(cset_comparison_base=1)
+    )
+    other: iris.cube.Cube = cubes.extract_cube(
+        iris.Constraint(
+            cube_func=lambda cube: "cset_comparison_base" not in cube.attributes
+        )
+    )
+
+    # Get spatial coord names.
+    base_lat_name, base_lon_name = get_cube_yxcoordname(base)
+    other_lat_name, other_lon_name = get_cube_yxcoordname(other)
+
+    # Ensure cubes to compare are on common differencing grid.
+    # This is triggered if either
+    #      i) latitude and longitude shapes are not the same. Note grid points
+    #         are not compared directly as these can differ through rounding
+    #         errors.
+    #     ii) or variables are known to often sit on different grid staggering
+    #         in different models (e.g. cell center vs cell edge), as is the case
+    #         for UM and LFRic comparisons.
+    # In future greater choice of regridding method might be applied depending
+    # on variable type. Linear regridding can in general be appropriate for smooth
+    # variables. Care should be taken with interpretation of differences
+    # given this dependency on regridding.
+    if (
+        base.coord(base_lat_name).shape != other.coord(other_lat_name).shape
+        or base.coord(base_lon_name).shape != other.coord(other_lon_name).shape
+    ) or (
+        base.long_name
+        in [
+            "eastward_wind_at_10m",
+            "northward_wind_at_10m",
+            "northward_wind_at_cell_centres",
+            "eastward_wind_at_cell_centres",
+            "zonal_wind_at_pressure_levels",
+            "meridional_wind_at_pressure_levels",
+            "potential_vorticity_at_pressure_levels",
+            "vapour_specific_humidity_at_pressure_levels_for_climate_averaging",
+        ]
+    ):
+        logging.debug(
+            "Linear regridding base cube to other grid to compute differences"
+        )
+        base = regrid_onto_cube(base, other, method="Linear")
+
+    def is_increasing(sequence: list) -> bool:
+        """Determine the direction of an ordered sequence.
+
+        Returns "increasing" or "decreasing" depending on whether the sequence
+        is going up or down. The sequence should already be monotonic, with no
+        duplicate values. An iris DimCoord's points fulfills this criteria.
```

Comment:
This docstring is incorrect, as it returns a boolean. (I see this is also an issue in `misc.difference`, so I'll fix it there via https://github.com/MetOffice/CSET/pull/1790.)

```suggestion
        Returns a boolean indicating that the values of a sequence are
        increasing. The sequence should already be monotonic, with no
        duplicate values. An iris DimCoord's points fulfils this criteria.
```